testing/test_stepwise.py

```diff
@@ -41,6 +41,14 @@ def test_success():
 """
     )
 
+    # customize cache directory so we don't use the tox's cache directory, which makes tests in this module flaky
```

Comment:
we ought to make this more general as i believe there are other bits that can and will affect that

its ok as a followup tho

Reply:
Sure, I will follow it up on #4378. 👍